demos/simple_pipeline/README.md

```diff
@@ -0,0 +1,118 @@
+# Kubeflow demo - Simple pipeline
+
+## Hyperparameter tuning and autoprovisioning GPU nodes
+
+This repository contains a demonstration of Kubeflow capabilities, suitable for
+presentation to public audiences.
+
+The base demo includes the following steps:
+
+1. [Setup your environment](#1-setup-your-environment)
+1. [Run a simple pipeline](#2-run-a-simple-pipeline)
+1. [Perform hyperparameter tuning](#3-perform-hyperparameter-tuning)
+1. [Run a better pipeline](#4-run-a-better-pipeline)
+
+## 1. Setup your environment
+
+Follow the instructions in
+[demo_setup/README.md](https://github.com/kubeflow/examples/blob/master/demos/simple_pipeline/demo_setup/README.md)
+to setup your environment and install Kubeflow with pipelines on an
+autoprovisioning GKE cluster.
+
+View the installed components in the GCP Console.
+*  In the
+[Kubernetes Engine](https://console.cloud.google.com/kubernetes)
+section, you will see a new cluster ${CLUSTER} with 3 `n1-standard-1` nodes
+*  Under
+[Workloads](https://console.cloud.google.com/kubernetes/workload),
+you will see all the default Kubeflow and pipeline components.
+
+Source the environment file and activate the conda environment for pipelines:
+
+```
+source kubeflow-demo-simple-pipeline.env
+source activate kfp
+```
+
+## 2. Run a simple pipeline
+
+Show the file `gpu-example-pipeline.py` as an example of a simple pipeline.
+
+Compile it to create a .tar.gz file:
+
+```
+./gpu-example-pipeline.py
+```
+
+View the pipelines UI locally by forwarding a port to the ml-pipeline-ui pod:
+
+```
+PIPELINES_POD=$(kubectl get po -l app=ml-pipeline-ui | \
+  grep ml-pipeline-ui | \
+  head -n 1 | \
+  cut -d " " -f 1 )
+kubectl port-forward ${PIPELINES_POD} 8080:3000
+```
+
+In the browser, navigate to `localhost:8080` and create a new pipeline by
+uploading `gpu-example-pipeline.py.tar.gz`. Select the pipeline and click
+"Create experiment." Use all suggested defaults.
+
+View the effects of autoscaling by watching the number of nodes.
+
+Select "Experiments" from the left-hand side, then "Runs". Click on the job to view
+the graph and watch it run.
+
+Notice the low accuracy.
+
+## 3. Perform hyperparameter tuning
```

Comment:
Added a link to the source of the gpu example file. I'm not sure how to apply a katib manifest using a pipeline step - @vicaire @qimingj do you know if that is supported?

Reply:
It is not supported in pipeline DSL, but it is supported in argo yaml since argo supports any K8s template, not just container spec. We can add Katib to DSL support (such as a KatibOp). In order to do that, ideally Katib's CRD should return some output in its job status (available via kubectl get) so argo can pick it up as output (with a JSON path to the field), and then the job output can be passed to downstream components. We should discuss this.